### PR TITLE
Show usage (short) help if command line parameter is wrong

### LINKED ISSUE #4986: Enhance PowerShell full command line help
- [ ] #5780 We should show the full/short command line help in Bash style if a parameter is wrong.
- [x] #4989 We should reformat the help in Bash style:
    - sort parameter list in alphabetical order.
    - replace long header on  powershell[.exe] [options] 
- [x] #4989 We should add more examples in line of the help.
- [x] #5275  Add "To get help type Help or Help followed by the command name, such as: Help Get-Process" in PowerShell banner bottom.

The Issue consider command line only  powershell -Help  (in bash or cmd.exe).

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary
Close #4986 

- Show short help after an error message:
```powershell
pwsh.exe -hhhhhhh
The argument '-hhhhhhh' is not recognized as the name of a script file. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

Usage: pwsh[.exe] [[-File] <filePath> [args]]
                  [-Command { - | <script-block> [-args <arg-array>]
                                | <string> [<CommandParameters>] } ]
                  [-ConfigurationName <string>] [-EncodedCommand <Base64EncodedCommand>]
                  [-ExecutionPolicy <ExecutionPolicy>] [-InputFormat {Text | XML}]
                  [-Interactive] [-NoExit] [-NoLogo] [-NonInteractive] [-NoProfile]
                  [-OutputFormat {Text | XML}] [-Version] [-WindowStyle <style>]

       pwsh[.exe] -h | -Help | -? | /?

PowerShell Online Help https://aka.ms/pscore6-docs

All parameters are case-insensitive.
```
- Show full help message with `pwsh -h`

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [NA] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [NA] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Comment by iSazonov** [comment]
CI AppVeyor temporary failed.

File: src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx [hunk 1]
```diff
@@ -123,6 +123,21 @@ Copyright (c) Microsoft Corporation. All rights reserved.
 
 https://aka.ms/pscore6-docs
 Type 'help' to get help.</value>
+  </data>
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx]: I don't like that we have a copy of the syntax.  If we make updates in the future, we need to manually ensure that both sections are updated.  Perhaps we can have the syntax (aka short help) just called `HelpSyntax` and the other one called `ExtendedHelp`.  Now change the code where full help would have been shown shows `HelpSyntax` followed by `ExtendedHelp`.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx]: If `ExtendedHelp` maybe second - `UsageHelp` or `ShortHelp`?

Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx]: `UsageHelp` sounds fine to me

Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/resources/ManagedEntranceStrings.resx]: Fixed.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs [hunk 2]
```diff
@@ -394,6 +394,10 @@ private void ShowHelp()
             Dbg.Assert(_helpText != null, "_helpText should not be null");
             _hostUI.WriteLine("");
             _hostUI.Write(_helpText);
+            if (_showExtendedHelp)
+            {
+                _hostUI.Write(ManagedEntranceStrings.ExtendedHelp);
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Shouldn't this show both `UsageHelp` and `ExtendedHelp`?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Line 396 already show `UsageHelp`.

Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs]: Got it!

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**